lib/Geo/S2/S2MultiPointRegion.cpp

```diff
@@ -20,31 +20,35 @@
 ///
 /// @author Simon Grätzer
 ////////////////////////////////////////////////////////////////////////////////
-#include "S2MultiPointRegion.h"
+
+#include "Geo/S2/S2MultiPointRegion.h"
+#include "Basics/Exceptions.h"
+
 #include <s2/s2cap.h>
 #include <s2/s2cell.h>
+#include <s2/s2latlng.h>
 #include <s2/s2latlng_rect.h>
 #include <s2/s2latlng_rect_bounder.h>
-#include <s2/util/coding/coder.h>
 
-S2MultiPointRegion::S2MultiPointRegion(std::vector<S2Point>* points)
-    : num_points_(points->size()), points_(nullptr) {
-  if (num_points_ > 0) {
-    points_ = new S2Point[num_points_];
-    memcpy(points_, points->data(), num_points_ * sizeof(S2Point));
-  }
-}
+namespace arangodb::geo {
 
-S2MultiPointRegion::S2MultiPointRegion(S2MultiPointRegion const* other)
-    : num_points_(other->num_points_), points_(nullptr) {
-  if (num_points_ > 0) {
-    points_ = new S2Point[num_points_];
-    memcpy(points_, other->points_, num_points_ * sizeof(S2Point));
+S2Point S2MultiPointRegion::GetCentroid() const noexcept {
```

Comment:
I know this algorithm has been in there before the PR (with an additional Normalize at the end). But I think it is seriously broken anyway. Imagine a bunch of points near the international date line, so this side, some on the other. Their Centroid according to this algorithm could easily be near the zero meridian at approximately the same lattitude, which is utter nonsense.
What should we do about this?

Reply:
Maybe we should compute centroid of S2Point(3 component vector)? It's still something strange in general, but at least we don't really suffer because meridian issue.

Reply:
It's strange because it will be "planar" centroid not on "sphere"

Reply:
I think the algorithm as presented does not make any sense. We could use the (normalized) vectors in R^3, take the centroid there and then project back to the sphere by normalizing. However, this has a singularity. If I would, for example, take two antipodal points, I would end up at (0,0,0) and can no longer produce a result. At the same time, the "centroid" isn't well defined anyway in this case. Maybe we should just not implement this, but this would break backward compatibility.
I do agree that the `Normalize` at the end of the old code is unnecessary.
How about the following: We add a comment saying that this algorithm does not make sense, that the concept is not well defined and that we just leave the old broken algorithm for backwards compatibility?
Furthermore, we say so in the documentation.
What do you think?

Reply:
I think in most cases it wouldn't be 0, 0, 0, so it can be normalized to unit length, and I think it can make some sense.

I understand it's not correct, but for near points it will pretty normal.
And at the same time, it's not suffer because lat closed to pole/or because 180 meridian.

Anyway I don't against any comments here, but if you allow I will add it in next pr, with other stuff

Reply:
Fine.